interfaces/builtin/browser.go

```diff
@@ -0,0 +1,223 @@
+# process drops this capability it can escape confinement, but once it
+# drops CAP_SYS_ADMIN we are ok.
+capability sys_admin,
+
+# All of these are for sanely dropping from root and chrooting
+capability chown,
+capability fsetid,
+capability setgid,
+capability setuid,
+capability sys_chroot,
+
+# User namespace sandbox
+owner @{PROC}/@{pid}/setgroups rw,
+owner @{PROC}/@{pid}/uid_map rw,
+owner @{PROC}/@{pid}/gid_map rw,
+`
+
+const browserConnectedPlugSecComp = `
+# Description: Can access various APIs needed by modern browers (eg, Google
+# Chrome/Chromium and Mozilla) and file paths they expect. This interface is
+# transitional and is only in place while upstream's work to change their paths
+# and snappy is updated to properly mediate the APIs.
+# Usage: reserved
+
+# for anonymous sockets
+bind
+listen
+
+# TODO: fine-tune when seccomp arg filtering available in stable distro
+# releases
+setpriority
+`
+
+const browserConnectedPlugSecCompWithSandbox = `
+# Policy needed only when using the chrome/chromium setuid sandbox
+chroot
+# TODO: fine-tune when seccomp arg filtering available in stable distro
+# releases
+setuid
+setgid
+
+# Policy needed for Mozilla userns sandbox
+unshare
+quotactl
+`
+
+type BrowserInterface struct{}
+
+func (iface *BrowserInterface) Name() string {
+	return "browser"
+}
+
+func (iface *BrowserInterface) SanitizeSlot(slot *interfaces.Slot) error {
+	return nil
+}
+
+func (iface *BrowserInterface) SanitizePlug(plug *interfaces.Plug) error {
+	if iface.Name() != plug.Interface {
+		panic(fmt.Sprintf("plug is not of interface %q", iface.Name()))
+	}
+
+	// It's fine if allow-browser-sandbox isn't specified, but it it is,
+	// it needs to be bool
+	if _, ok := plug.Attrs["allow-browser-sandbox"]; ok {
+		if _, ok = plug.Attrs["allow-browser-sandbox"].(bool); !ok {
+			return fmt.Errorf("browser plug requires bool with 'allow-browser-sandbox'")
+		}
+	}
+
+	return nil
+}
+
+func (iface *BrowserInterface) ConnectedSlotSnippet(plug *interfaces.Plug, slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	switch securitySystem {
+	case interfaces.SecurityAppArmor, interfaces.SecuritySecComp, interfaces.SecurityDBus, interfaces.SecurityUDev, interfaces.SecurityMount:
+		return nil, nil
+	default:
+		return nil, interfaces.ErrUnknownSecurity
+	}
+}
+
+func (iface *BrowserInterface) PermanentSlotSnippet(slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+
+	switch securitySystem {
+	case interfaces.SecurityAppArmor, interfaces.SecuritySecComp, interfaces.SecurityDBus, interfaces.SecurityUDev, interfaces.SecurityMount:
+		return nil, nil
+	default:
+		return nil, interfaces.ErrUnknownSecurity
+	}
+}
+
+func (iface *BrowserInterface) ConnectedPlugSnippet(plug *interfaces.Plug, slot *interfaces.Slot, securitySystem interfaces.SecuritySystem) ([]byte, error) {
+	allow_browser_sandbox := false
```

Comment:
This will do all of the logic here:
```
allowSandbox, _ := plug.Attrs["allow-sandbox"].(bool)
```
Please note we use camelCase rather than under_lines.